File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 1, max_pr = 0;
        while (r < prices.size()) {
            if (prices[l] < prices[r]) {
                max_pr = max(max_pr, prices[r] - prices[l]);
            } else {
                l = r;
            }
            r++;
        }
        return max_pr;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

def getHeight(root):
    if not root:
        return 0
    return 1 + max(getHeight(root.left), getHeight(root.right))

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True

        q = deque()
        q.append(root)
        while q:
            for i in range(len(q)):
                node = q.popleft()
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

                if abs(getHeight(node.left) - getHeight(node.right)) > 1:
                    return False

        return True